clients/unshaded/src/main/java/tachyon/client/FileOutStream.java

```diff
@@ -123,7 +123,8 @@ public void close() throws IOException {
       } catch (IOException ioe) {
         if (mWriteType.isMustCache()) {
           LOG.error(ioe.getMessage(), ioe);
-          throw new IOException("Fail to cache: " + mWriteType, ioe);
+          throw new IOException("Fail to cache: " + mWriteType + ", message: " + ioe.getMessage(),
+              ioe);
```

Comment:
You are right. However, the exception is not very descriptive. For example, if anything fails in an FileOutStream, I get:

java.io.IOException: Fail to cache: MUST_CACHE

However, by adding the message here, I know what's going on:

java.io.IOException: Fail to cache: MUST_CACHE, message: java.io.IOException: Decode error. Possible Client/DataServer version incompatibility.

I feel like the second one is more obvious what happened, instead of having to go through a log. (On jenkins, I don't even think the logs are accessible).

What do you think?